### ISSUE TITLE
Need sample or doc for IPackageEncoder

### ISSUE THREAD

**Issue by ishowfun** [kerryjiang/SuperSocket#471]
pls write a sample or doc for IPackageEncoder
or pls recommand a third-party extension of IBufferWriter 

and why you replace ArraySegment by IBufferWriter?

**Comment by kerryjiang** [comment]
Yes, I will add more samples about the usage of IPackageEncoder.

For now, you can take a look at the implementation of WebSocket
https://github.com/kerryjiang/SuperSocket/blob/master/src/SuperSocket.WebSocket/WebSocketEncoder.cs

and the usage of BinaryPrimitives

https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1

https://github.com/SciSharp/dotnet-mysql-replication/blob/c66ea5c5eb1df639bd9598e7eef119fdc9a62b0d/src/SciSharp.MySQL.Replication/ReplicationClient.cs#L184-L220
